### PR TITLE
[SYCL][ABI-break] Remove the workaround for release of auxiliary buffers

### PR DESCRIPTION
**Pull request by sergey-semenov**
Remove the thread local variable workaround for deferred release of auxiliary resources by passing the required information from reduction headers.

### CODE REVIEW

File: sycl/include/sycl/handler.hpp [hunk 1]
```diff
@@ -441,13 +441,26 @@ class __SYCL_EXPORT handler {
...
   /// the command group finishes the work on device/host.
   ///
   /// @param ReduObj is a pointer to object that must be stored.
   void addReduction(const std::shared_ptr<const void> &ReduObj);
 
+  /// Saves buffers created by handling reduction feature in handler and marks
+  /// them as internal. They are then forwarded to command group and destroyed
+  /// only after the command group finishes the work on device/host.
+  ///
+  /// @param ReduObj is a pointer to object that must be stored.
```
Comment (KseniyaTikhomirova) [inline sycl/include/sycl/handler.hpp:455]: ReduBuf

**Review by KseniyaTikhomirova — COMMENTED** [review]
windows symbols update could be also needed

**Comment by sergey-semenov** [comment]
@KseniyaTikhomirova Could you please review this change?

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

**Pull request merged**